scripts/pre-commit/validate_migration_scripts.py

```diff
@@ -0,0 +1,189 @@
+import subprocess
+import sys
+import re
+import logging
+from pathlib import Path
+import typing as tp
+from datetime import datetime
+
+logging.basicConfig(level=logging.INFO, format="%(levelname)s - %(message)s")
+logger = logging.getLogger("validate_migration_scripts")
+
+CODE_EXTENSION = ".sql"
+BOOTSTRAP_FILE = "bootstrap.sql"
+METADATA_STORE_SCRIPTS_DIR = "metadata_store_scripts"
+MIGRATION_SCRIPT_COMMENT = "-- source: "
+MIGRATION_SCRIPT = "scripts/migrations.sh"
+
+
+class MigrationValidationError(Exception):
+    """Raised when a migration validation check fails."""
+
+
+def is_code_file(filepath: str) -> bool:
+    return Path(filepath).suffix == CODE_EXTENSION
+
+
+def get_project_root() -> Path:
+    project_root = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        capture_output=True,
+        text=True,
+        check=True
+    )
+    return Path(project_root.stdout.strip())
+
+
+def parse_date(version_str: str) -> datetime:
+    date_str = ".".join(version_str[1:].split(".")[:3])
+    return datetime.strptime(date_str, "%Y.%m.%d")
+
+
+def check_duplicate_migration_versions(new_migration_files: tp.List, old_migration_files: tp.List) -> None:
+    """
+    Check if there are any duplicate migration versions
+    1. Between new and old migration files.
+    2. Between new migration files.
+    Migration Version Example - V2025.02.24.0
+    """
+    new_versions = [file.split("_")[0] for file in new_migration_files if file != BOOTSTRAP_FILE]
+    old_versions = [file.split("_")[0] for file in old_migration_files]
+
+    if duplicates := set(new_versions) & set(old_versions):
+        raise MigrationValidationError(
+            f"Duplicate migration versions found: {duplicates}"
+        )
+
+    new_migration_files_set = set(new_migration_files)
+    if duplicates := [file for file in new_migration_files_set if new_versions.count(file.split("_")[0]) > 1]:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found in new migration files: {duplicates}"
+        )
+
+
+def check_new_migration_file_format(new_migration_files: tp.List) -> None:
+    """
+    Check if the new migration files have a date format in their names.
+    """
+    for file in new_migration_files:
+        if file == BOOTSTRAP_FILE:
+            continue
+        if not re.match(r"^V\d{4}.\d{2}.\d{2}.\d{1,2}__.+\.sql$", file):
```

Comment:
question: why are we specifically choosing to allow two digit revisions? we don't currently have any two digit revisions, and it seems unnecessary to add support for this since that would mean a PR has more than 10 separate migrations added